app/lib/actions/pulp/abstract_async_task_group.rb

```diff
@@ -31,6 +32,23 @@ def run(event = nil)
         end
       end
 
+      def cancel!
+        cancel
+        self.external_task = poll_external_task
+        # We suspend the action and the polling will take care of finding
+        # out if the cancelling was successful
+        suspend unless done?
+      end
+
+      def cancel
+        group = output['pulp_task_group']
+        if group
+          task_resource.cancel(group['id'])
+        end
```

Comment:
ya i see what you mean. I have never seen an issue around this, so it may be overly defensive to attempt to handle this case. 👍 to leaving it as-is